oxd-server/src/main/java/org/gluu/oxd/server/persistence/providers/GluuPersistenceConfiguration.java

```diff
@@ -0,0 +1,95 @@
+package org.gluu.oxd.server.persistence.providers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang.StringUtils;
+import org.gluu.oxd.common.Jackson2;
+import org.gluu.oxd.common.PersistenceConfigKeys;
+import org.gluu.oxd.server.OxdServerConfiguration;
+import org.gluu.oxd.server.Utils;
+import org.gluu.oxd.server.persistence.configuration.GluuConfiguration;
+import org.gluu.util.exception.ConfigurationException;
+import org.gluu.util.security.PropertiesDecrypter;
+import org.gluu.util.security.StringEncrypter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Properties;
+
+public class GluuPersistenceConfiguration {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JDBCPersistenceProvider.class);
+
+    private OxdServerConfiguration configuration;
+    private Properties connectionProperties;
+
+    public GluuPersistenceConfiguration(OxdServerConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    public Properties getPersistenceProps() {
+        try {
+            GluuConfiguration gluuConfiguration = asGluuConfiguration(this.configuration);
+            validate(gluuConfiguration);
+
+            //read persistence `base` file
+            Properties props = Utils.loadPropertiesFromFile(gluuConfiguration.getType(), null);
+            //read persistence `connection` file
+            props = Utils.loadPropertiesFromFile(gluuConfiguration.getConnection(), props);
+            this.connectionProperties = props;
+            //read salt file
+            Properties saltProps = Utils.loadPropertiesFromFile(gluuConfiguration.getSalt(), null);
+            return preparePersistanceProperties(saltProps.getProperty(PersistenceConfigKeys.EncodeSalt.getKeyName()));
+
+        } catch (Exception e) {
+            throw new IllegalStateException("Error starting GluuPersistenceService", e);
+        }
+    }
+
+    protected Properties preparePersistanceProperties(String cryptoConfigurationSalt) {
+
+        Properties decryptedConnectionProperties;
+        try {
+            decryptedConnectionProperties = PropertiesDecrypter.decryptAllProperties(StringEncrypter.defaultInstance(), this.connectionProperties, cryptoConfigurationSalt);
+        } catch (StringEncrypter.EncryptionException ex) {
+            throw new ConfigurationException("Failed to decript configuration properties", ex);
+        }
+
+        return decryptedConnectionProperties;
+    }
+
+    public static GluuConfiguration asGluuConfiguration(OxdServerConfiguration configuration) {
+        try {
+            JsonNode node = configuration.getStorageConfiguration();
+            if (node != null) {
+                return Jackson2.createJsonMapper().treeToValue(node, GluuConfiguration.class);
+            }
+        } catch (Exception e) {
+            LOG.error("Failed to parse GluuConfiguration.", e);
+        }
+        return new GluuConfiguration();
```

Comment:
Better `Optional<GluuConfiguration>` and then `Optional.absetnt()`. Otherwise caller of this method have no idea that it can  blank/empty object.

Reply:
Done